### PR TITLE
Remove obsolete pixel filter methods

### PR DESCRIPTION
**Pull request by Mango-3**
- Removes pixel filter which had been removed from appleseed-core when filter importance sampling was introduced. 

The removed legacy filter are all mapped now to Blackman-Harris, the supported pixel filter mappings are unaffected. Removed pixel filter will show up in the UI marked legacy (see screenshot)
Capture

### CODE REVIEW

File: src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp [hunk 1]
```diff
@@ -1725,15 +1725,15 @@ namespace
...
-          case 0:  return "blackman-harris";
-          case 1:  return "box";
-          case 2:  return "catmull";
-          case 3:  return "bspline";
-          case 4:  return "gaussian";
-          case 5:  return "lanczos";
-          case 6:  return "mitchell";
-          case 7:  return "triangle";
-          default: return "box";
+        case 0:  return "blackman-harris";
```
Comment (usakhelo) [inline src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp]: I think two space indentation is missing here.

File: src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp [hunk 2]
```diff
@@ -195,11 +195,11 @@ namespace
     {
         { IDS_RENDERERPARAMS_FILTER_TYPE_1,             L"Blackman-Harris" },
         { IDS_RENDERERPARAMS_FILTER_TYPE_2,             L"Box" },
-        { IDS_RENDERERPARAMS_FILTER_TYPE_3,             L"Catmull-Rom Spline" },
-        { IDS_RENDERERPARAMS_FILTER_TYPE_4,             L"Cubic B-spline" },
+        { IDS_RENDERERPARAMS_FILTER_TYPE_3,             L" Catmull-Rom Spline - Legacy" },
```
Comment (usakhelo) [inline src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp]: One space indentation is deliberate I suppose?

File: src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp (hunk 1 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp]: True, the auto indent struck again.

File: src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp (hunk 2 above)
Comment (Mango-3) [reply src/appleseed-max-impl/appleseedrenderer/appleseedrenderer.cpp]: It is intentional. My idea was to have the legacy filter entries slightly offset in the selection menu (see screenshot above) to make it visually more distinct.

File: src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp [hunk 3]
```diff
@@ -1727,13 +1727,13 @@ namespace
         {
           case 0:  return "blackman-harris";
           case 1:  return "box";
-          case 2:  return "catmull";
-          case 3:  return "bspline";
+          case 2:  return "blackman-harris"; // legacy catmul mapped to blackman-harris
```
Comment (dictoon) [inline src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp]: Typo: *catmull*

**Comment by Mango-3** [comment]
> So if I understand correctly, on an older scene that uses one of the removed filter, the filter combo will show the name of that removed filter (with a "legacy" tag) but under the hood it will use Blackman-Harris

Yes, this is the case. It keeps the order of the supported pixel filter intact when loading legacy scene files. The obsolete ones are all remapped to Blackman-Harris. 
The legacy tag should user discourage using it for new scenes.

**Pull request merged**